SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplicationSubobject.cpp

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialTestPropertyReplicationSubobject.h"
+#include "Kismet/GameplayStatics.h"
+#include "ReplicatedTestActorSubobject.h"
+
+/**
+ * "This tests that the data debug mode detects property changes on non-auth servers for subobjects.
+ * This test contains 2 Servers and 2 Client workers.
+ *
+ * The flow is as follows:
+ * - Setup:
+ *  - The authorative server spawns one ReplicatedTestActorSubobject (in it's OnAuthorityGained it will spawn a subobject of type
+ * ReplicatedTestActor )
+ * - Test:
+ *  - All workers check that they can see exactly 1 ReplicatedTestActor and 1 ReplicatedTestActorSubobject.
+ *  - The authorative server changes the replicated property of the replicated subobject.
+ *  - All workers check that the replicated property of the replicated subobject has changed.
+ *  - The non-auth server changes the the replicated property of the replicated subobject which generates two expected errors.
+ *  - Auth server check that the replicated property of the replicated subobject has not changed.
+ * - Clean-up:
+ *  - ReplicatedTestActor is destroyed using the RegisterAutoDestroyActor helper function.
+ */
+
+ASpatialTestPropertyReplicationSubobject::ASpatialTestPropertyReplicationSubobject()
+	: Super()
+{
+	Author = "Victoria Bloom";
+	Description = TEXT("This tests that the data debug mode detects property changes on non-auth servers for replicated subobjects.");
+}
+
+void ASpatialTestPropertyReplicationSubobject::PrepareTest()
+{
+	Super::PrepareTest();
+
+	if (HasAuthority())
+	{
+		// Subobject property change - expect an actor level error and a property level error
+		AddExpectedLogError(TEXT("Changed actor without authority with name ReplicatedTestActorSubobject_0 of type "
+								 "ReplicatedTestActorSubobject, property changed without authority was ReplicatedSubActor!"),
+							1, true);
+
+		AddExpectedLogError(TEXT("Changed actor without authority with name ReplicatedTestActor_0 of type ReplicatedTestActor, property "
+								 "changed without authority was TestReplicatedProperty!"),
+							1, true);
+	}
+
+	AddStep(
+		TEXT("The auth server spawns one ReplicatedTestActorSubobject"), FWorkerDefinition::Server(1), nullptr,
+		[this]() {
+			TestActor = GetWorld()->SpawnActor<AReplicatedTestActorSubobject>(FVector(0.0f, 0.0f, 50.0f), FRotator::ZeroRotator,
+																			  FActorSpawnParameters());
+			RegisterAutoDestroyActor(TestActor);
+
+			FinishStep();
+		},
+		nullptr, 5.0f);
+
+	AddStep(
```

Comment:
As with other test comments, I am unsure if we should add a step for actor ready.